Compilation failure with GCC 11.x

The release of GCC 11 introduces a new warning `array-parameter` causing the build of mirage-crypto's C stubs to fail:
```
native/des_generic.c:404:31: error: argument 1 of type ‘unsigned char[16]’ with mismatched bound [-Werror=array-parameter=]
  404 | void mc_des2key(unsigned char hexkey[16], short mode) /* stomps on Kn3 too */
      |                 ~~~~~~~~~~~~~~^~~~~~~~~~
In file included from native/des_generic.c:19:
native/des_generic.h:67:24: note: previously declared as ‘unsigned char *’
   67 | extern void mc_des2key(unsigned char *, short);
      |                        ^~~~~~~~~~~~~~~
native/des_generic.c:536:31: error: argument 1 of type ‘unsigned char[24]’ with mismatched bound [-Werror=array-parameter=]
  536 | void mc_des3key(unsigned char hexkey[24], short mode)
      |                 ~~~~~~~~~~~~~~^~~~~~~~~~
In file included from native/des_generic.c:19:
native/des_generic.h:126:24: note: previously declared as ‘unsigned char *’
  126 | extern void mc_des3key(unsigned char *, short);
      |                        ^~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

--- a/src/native/des_generic.h
+++ b/src/native/des_generic.h
@@ -64,7 +64,7 @@ extern void mc_des(unsigned char *, unsigned char *);
 #ifdef D2_DES
 
 #define desDkey(a,b)	mc_des2key((a),(b))
-extern void mc_des2key(unsigned char *, short);
+extern void mc_des2key(unsigned char [16], short);
 /*		      hexkey[16]     MODE
  * Sets the internal key registerS according to the hexadecimal
  * keyS contained in the 16 bytes of hexkey, according to the DES,
@@ -123,7 +123,7 @@ extern void mc_cp2key(unsigned long *);
 #define useDkey(a)	mc_use3key((a))
 #define cpDkey(a)	mc_cp3key((a))
 
-extern void mc_des3key(unsigned char *, short);
+extern void mc_des3key(unsigned char [24], short);
 /*		      hexkey[24]     MODE
  * Sets the internal key registerS according to the hexadecimal
  * keyS contained in the 24 bytes of hexkey, according to the DES,

